openedx/core/djangoapps/user_api/preferences/tests/test_api.py

```diff
@@ -181,6 +181,34 @@ def test_update_user_preferences(self):
             "new_value"
         )
 
+    def test_update_user_preferences_with_username(self):
```

Comment:
@nasthagiri 
Unfortunately, that is not the case with current implementation and there is a test case  too for Staff User not Authorized https://github.com/edx/edx-platform/blob/jia/MA-1281/openedx/core/djangoapps/user_api/preferences/tests/test_api.py#L230
And the document also says so https://github.com/edx/edx-platform/blob/jia/MA-1281/openedx/core/djangoapps/user_api/preferences/views.py#L65

Staff user only has GET rights for other user's preferences. https://github.com/edx/edx-platform/blob/jia/MA-1281/openedx/core/djangoapps/user_api/preferences/views.py#L50

Let me know if we need to update this behaviour and allow staff user to access other user's preference info.